Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -190,6 +197,11 @@ def __init__(self,
         # splitting Gradient
         self.split = split
 
+        # warm-start
+        self.warmstart  = warmstart
+        if self.warmstart:
+            self.hasstarted = False
```

Comment:
I'm not a fan of this. I think we could make it with a `property` along these lines:

```python
@property
def p2(self):
    if self._p2 is not None:
        return self._p2
    else:
        p2 = self.gradient.range_geometry().allocate(0)
        if self.warmstart:
            self.p2 = p2
        return p2

@p2.setter
def p2(self, value):
    # possibly one would like to do some checks?
    self._p2 = value
            

```

Then we need to update `proximal`

Reply:
Thanks- have changed this